File: Practica1/S1E2/lib/Carrera.rb
require_relative "Bicicleta.rb"

class Carrera
  @bicicletas = Array.new
  @numBicicletas = 0
  @retirados = 0
  
  def getBicicletas
    return bicicletas
  end
  
  def iniciar
    # Poner tiempo de retiro
    Bicicleta.setRetiro(Random.rand(2))
    
    i = 0
    # Iniciar
    for b in @bicicletas
      i = i + 1
      b.run
    end
  end
  
  def getRetirados
    return @retirados
  end
  
  def getNumBicicletas
    return @numBicicletas
  end
  
  def setBicicletas(bicis)
    @bicicletas = bicis
  end
end

File: Practica1/S1E1/src/carrerasbicicletas/Carrera.java
package carrerasbicicletas;
import java.util.ArrayList;
import java.util.Random;

public abstract class Carrera {
    public ArrayList<Bicicleta> bicicletas;
    public int numBicicletas;
    
    public ArrayList<Bicicleta> getBicicletas(){
        return bicicletas;
    }
    
    public void iniciarHebras(int retirados){
        // Poner tiempo de retiro
        Random rand = new Random();
        Bicicleta.settRetiro(rand.nextInt(59999));
        
        // Iniciar hebras
        for (int i = 0 ; i < numBicicletas ; i++){
            bicicletas.get(i).start();
            
            if (i < retirados){
                bicicletas.get(i).setRetirada(true);
            }
        }
    }
    
    public abstract void iniciarCarrera();
}
